### DISCUSSION TITLE
Failed to read fields from ECMWF ensemble grib1 files

### DISCUSSION THREAD

**Discussion by BinbinZhou-NOAA** [dtcenter/METplus#1066] [Configuration] [answered]
I am testing to use ecmwf's original grib1 format files as input to run ensemble_stat
instead of first converting them to NCEP standard grib1/2 files. 
Use 500bm geopotential height as example,  wgrib one of files show following: 
7:587412:d=21080100:GH:kpds5=156:kpds6=100:kpds7=500:TR=0:P1=48:P2=0:TimeU=1:500 mb:48hr fcst:type=Perturbed forecast 1:NAve=0 
So I setup the forecast and obs field in the config as   
  field = [
      {
         name       = "GH";
         level      = "P500";
      }
  ]; 

But error message of running ensemble_stat  is
ERROR  : VarInfoGrib::add_grib_code() -> unrecognized GRIB1 field abbreviation 'GH' for table version 2 

But using plot_data_plane 
plot_data_plane ecme.ens01.t00z.grid3.f048.grib1 plot.ps 'name="GH"; level="P500";' -v 3 
can correctly generate the plot.  
I don't know what is the problem is.
Mt testing directory on VENUS:
/gpfs/dell2/emc/modeling/noscrub/Binbin.Zhou/met/ecme_orig_grib1
the running script is run_met.sh

Thanks!

Binbin 
EMC/NCEP

**Comment by malloryprow** [comment]
Hi Binbin,

I think you can use the configuration option "GRIB1_ptv". From https://met.readthedocs.io/en/latest/Users_Guide/config_options.html:

> The “GRIB1_ptv” entry is an integer specifying the GRIB1 parameter table version number.

I think the ECMWF data uses parameter table version number 128, but you would need to used wgrib to be sure. If it is 128, I think the config would look like...

field = [
{
name = "GH";
level = "P500";
GRIB1_ptv = 128;
}
];

**Reply by BinbinZhou-NOAA** [reply]
Mallory,

   It seems still not working, error says
ERROR  : VarInfoGrib::add_grib_code() -> unrecognized GRIB1 field abbreviation 'GH' for table version 128 

Binbin

**Reply by malloryprow** [reply]
Darn, bummer!

**Reply by BinbinZhou-NOAA** [reply]
Mallory,

  Your suggestion reminds me using GRIB1_code instead of using name to specify
the field . For Geopotential Height
set GRIB1_code = 156 
Now it work!

Thanks!

Binbin

**Reply by BinbinZhou-NOAA** [reply]
Further issue:

Putting the above setting is working in MET configuation file (run outside of METplus), but put this setting in METplus conf file like following

ENS_VAR1_LEVELS = P500
ENS_VAR1_OPTIONS = GRIB1_code = 156

I still get error message:
08/05 19:05:45.516  metplus (command_builder.py:255) ERROR: (ensemble_stat_wrapper.py:467) Could not build field info for fcst, obs, or ens  

Is seems the name of a field can not be missed  in METplus conf file, but can be in MET config file.  I do see the ECMWF version 128 table file grib1_ecmwf_128_98.txt is listed in MET_10.0's table directoty share/met/table_files, and "GH" is  string for "Height",  but why it  can not be recognized?   

Any suggestion?

Thanks!

Binbin

**Reply by JohnHalleyGotway** [reply]
Binbin, I was able to log on and take a look at the data that Julie pulled down. I agree that this behavior is problematic.

Unfortunately, I suspect there is an issue in the GRIB table handling logic which will require a bugfix. I am out of the office this week on vacation, but will return on Monday 8/16.

**Comment by BinbinZhou-NOAA** [comment]
John,

    No problem, enjoy your vacation.

Binbin


On Tue, Aug 10, 2021 at 8:30 PM johnhg ***@***.***> wrote:

> Binbin, I was able to log on and take a look at the data that Julie pulled
> down. I agree that this behavior is problematic.
>
> Unfortunately, I suspect there is an issue in the GRIB table handling
> logic which will require a bugfix. I am out of the office this week on
> vacation, but will return on Monday 8/16.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/1066#discussioncomment-1156505>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFPMQA5IYS3B436FBS3T4HABPANCNFSM5BUK5QJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 

Binbin Zhou

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

**Comment by JohnHalleyGotway** [comment, marked as answer]
@BinbinZhou-NOAA, I see what's going on here and wrote up this bugfix issue for dtcenter/MET#1879.

However, including the following in your Ensemble-Stat config file should enable the current version to work. Note that I'm explicitly specifying a value for GRIB1_center:
```
obs = {
   field = [
      {  
         name         = "GH";
         level        = "P500";
         GRIB1_ptv    = 128;
         GRIB1_center = 98;
     }
   ];
}
```
Please try using that and let me know if that gets Ensemble-Stat running for you.

**Reply by BinbinZhou-NOAA** [reply]
John,

   I have tested your suggested setting. Yes  it works for current MET_10.0.0.   
Now further queston. 

Thanks!

Binbin

**Reply by JohnHalleyGotway** [reply]
Thanks for confirming Binbin. I submitted a pull request to change the behavior in the next official release and bugfix release to make things easier. And it also makes the error message more clear. I'll go ahead and mark this discussion as being answered.
